src/components/ShareDialog.jsx

```diff
@@ -36,51 +36,55 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-export default function ShareDialog(props) {
+export default function ShareDialog() {
+  const location = useLocation();
+  const { showShareDialog, shareTitle, shareURL } = location.state
+    ?.shareDialog || {
+    showShareDialog: false,
+  };
+
+  const history = useHistory();
+  const closeDialog = () =>
+    history.replace({
+      state: { shareDialog: { shareTitle, shareURL, showShareDialog: false } },
```

Comment:
And then I think this would either be:
```suggestion
      state: { showShareDialog: false },
```
or if the `shareData` state isn't preserved above then:
```suggestion
      state: { shareData },
```

I think it's the latter.

Reply:
yup the latter works.